### ISSUE TITLE
Danger should support older Babel versions

### ISSUE THREAD

**Issue by sajjadzamani** [danger/danger-js#781]
# Problem
I wanted to import other modules using the es6 `import` in my Dangerfile. I have Babel and Jest setup in my project so I expected the Dangerfile to be transpiled. However, Danger did not transpile my Dangerfile and the `import` statements threw errors.
### Project Dependencies
```JS
"devDependencies": {
    "babel-core": "^6.26.3",
    "babel-eslint": "7.1.1",
    "babel-jest": "^23.6.0",
    "babel-runtime": "^6.26.0",
    "danger": "^6.1.9",
    "danger-plugin-jest": "^1.1.0",
    ...
  }
```
### Debug Output
```
2018-12-06T17:32:02.854Z danger:runner Started Danger runner with

2018-12-06T17:32:02.865Z danger:runner Got STDIN for Danger Run

2018-12-06T17:32:02.867Z danger:jsonToDSL Creating pr DSL from JSON

2018-12-06T17:32:02.924Z danger:localGetDiff > git diff master...HEAD

2018-12-06T17:32:02.939Z danger:localGetDiff > git log master...HEAD --pretty=format:{ "sha": "%H", "parents": "%p", "author": {"name": "%an", "email": "%ae", "date": "%ai" }, "committer": {"name": "%cn", "email": "%ce", "date": "%ci" }, "message": "%f"},

2018-12-06T17:32:02.958Z danger:runner Evaluating dangerfile.js

2018-12-06T17:32:02.959Z danger:transpiler:setup Does not have TypeScript set up

2018-12-06T17:32:02.960Z danger:transpiler:setup Does not have Babel set up

2018-12-06T17:32:02.960Z danger:inline_runner Started parsing Dangerfile:  dangerfile.js

Unable to evaluate the Dangerfile
 dangerfile.js:4
import jest from "danger-plugin-jest";
^^^^^^

SyntaxError: Unexpected token import
    at createScript (vm.js:56:10)
    at Object.runInThisContext (vm.js:97:10)
    at Module._compile (module.js:542:28)
    at Object.requireFromString [as default] (/Users/sajjad/Documents/TapSource/tap-react/node_modules/require-from-string/index.js:28:4)
    at Object.<anonymous> (/Users/sajjad/Documents/TapSource/tap-react/node_modules/danger/distribution/runner/runners/inline.js:144:63)
    at step (/Users/sajjad/Documents/TapSource/tap-react/node_modules/danger/distribution/runner/runners/inline.js:32:23)
    at Object.next (/Users/sajjad/Documents/TapSource/tap-react/node_modules/danger/distribution/runner/runners/inline.js:13:53)
    at /Users/sajjad/Documents/TapSource/tap-react/node_modules/danger/distribution/runner/runners/inline.js:7:71
    at __awaiter (/Users/sajjad/Documents/TapSource/tap-react/node_modules/danger/distribution/runner/runners/inline.js:3:12)
    at Object.exports.runDangerfileEnvironment (/Users/sajjad/Documents/TapSource/tap-react/node_modules/danger/distribution/runner/runners/inline.js:105:136)
```
# Investigation
I did some investigation into Danger and realized that Danger checks for the `@babel/core` module to detect if the project has a Babel dependency. However, this module was named `babel-core` in versions prior to Babel 7 which is why my Dangerfile does not get transpiled. It would be great if older versions of Babel were also supported. If it was intended to only support Babel `7.x.x` this should be documented.

**Comment by sajjadzamani** [comment]
Great! I will look into it.

**Comment by sajjadzamani** [comment]
@orta I have created a PR for this. Overall, I don't know if this change is valuable. Personally, I only needed Babel for `import` statements which not a huge deal. The main reason I created this issue is that I was wondering why my Dangerfile was not being transpiled when according to docs it was supposed to. So if the PR doesn't end up getting merged, then I'll just update the docs to mention that only Babel 7 is supported. I also saw that `@babel/polyfill` is needed, correct?
